src/Resources/config/commands.php

```diff
@@ -31,15 +31,15 @@
                 new ReferenceConfigurator('validator'),
             ])
 
-        // NEXT_MAJOR: Remove "doctrine" argument.
+        // NEXT_MAJOR: Remove the "setRegistry" call.
         ->set(GenerateObjectAclCommand::class, GenerateObjectAclCommand::class)
             ->public()
             ->tag('console.command')
             ->args([
                 new ReferenceConfigurator('sonata.admin.pool'),
                 [],
-                (new ReferenceConfigurator('doctrine'))->nullOnInvalid(),
             ])
+            ->call('setRegistry', [(new ReferenceConfigurator('doctrine'))->nullOnInvalid()])
```

Comment:
Since you made the change in the command ; shouldn't you remove the call too ?

Reply:
IMO not in this PR that should be just to merge up the `3.x` branch (the extra commit I added was to fix PHPStan and Psalm issues).